Source/Core/Core/HW/DVD/AMMediaboard.cpp

```diff
@@ -0,0 +1,1831 @@
+        {
+          buffer = (u8*)(s_network_buffer + off - NetworkBufferAddress4);
+        }
+        else
+        {
+          PanicAlertFmt("RECV: Buffer overrun:{0} {1} ", off, len);
+        }
+
+        const int ret = recv(fd, (char*)buffer, len, 0);
+        const int err = WSAGetLastError();
+
+        NOTICE_LOG_FMT(AMMEDIABOARD_NET, "GC-AM: recv( {}, 0x{:08x}, {} ):{} {}\n", fd, off, len,
+                       ret, err);
+
+        s_media_buffer[1] = s_media_buffer[8];
+        s_media_buffer_32[1] = ret;
+        break;
+      }
+      case AMMBCommand::Send:
+      {
+        const u32 fd = s_sockets[SocketCheck(s_media_buffer_32[2])];
+        u32 off = s_media_buffer_32[3];
+        const u32 len = s_media_buffer_32[4];
+
+        if (off >= NetworkBufferAddress3 && off < NetworkBufferAddress3 + sizeof(s_network_buffer))
+        {
+          off -= NetworkBufferAddress3;
+        }
+        else
+        {
+          ERROR_LOG_FMT(AMMEDIABOARD_NET, "GC-AM: send(error) unhandled destination:{:08x}\n", off);
+        }
+
+        const int ret = send(fd, (char*)(s_network_buffer + off), len, 0);
+        const int err = WSAGetLastError();
+
+        NOTICE_LOG_FMT(AMMEDIABOARD_NET, "GC-AM: send( {}({}), 0x{:08x}, {} ): {} {}\n", fd,
+                       s_media_buffer_32[2], off, len, ret, err);
+
+        s_media_buffer[1] = s_media_buffer[8];
+        s_media_buffer_32[1] = ret;
+        break;
+      }
+      case AMMBCommand::Socket:
+      {
+        // Protocol is not sent
+        const u32 af = s_media_buffer_32[2];
+        const u32 type = s_media_buffer_32[3];
+
+        const SOCKET fd = socket_(af, type, IPPROTO_TCP);
+
+        NOTICE_LOG_FMT(AMMEDIABOARD_NET, "GC-AM: socket( {}, {}, IPPROTO_TCP ):{}\n", af, type, fd);
+
+        s_media_buffer[1] = 0;
+        s_media_buffer_32[1] = fd;
+        break;
+      }
+      case AMMBCommand::Select:
+      {
+        u32 fd = s_sockets[SocketCheck(s_media_buffer_32[2] - 1)];
+
+        // BUG: NAMCAM is hardcoded to call this with socket ID 0x100 which might be some magic
+        // thing? A valid value is needed so we use the socket from the connect.
+
+        if (AMMediaboard::GetGameType() == MarioKartGP2)
+        {
+          if (s_media_buffer_32[2] == 256)
+          {
+            fd = s_namco_cam;
+          }
+        }
+
+        fd_set* readfds = nullptr;
+        fd_set* writefds = nullptr;
+        fd_set* exceptfds = nullptr;
+        timeval* timeout = nullptr;
+
+        // Only one of 3, 4, 5 is ever set alongside 6
+        if (s_media_buffer_32[3] && s_media_buffer_32[6])
+        {
+          const u32 ROffset = s_media_buffer_32[6] - NetworkCommandAddress2;
+          readfds = (fd_set*)(s_network_command_buffer + ROffset);
+          FD_ZERO(readfds);
+          FD_SET(fd, readfds);
+
+          timeout =
+              (timeval*)(s_network_command_buffer + s_media_buffer_32[3] - NetworkCommandAddress2);
```

Comment:
Looks like this is violating the strict aliasing rule. Same problem in the alternate paths.
`memcpy` or `Common::BitCastPtr` can be used.